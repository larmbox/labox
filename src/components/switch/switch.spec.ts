import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { LSwitch } from '~/index';

test('mount component', async () => {
  expect(LSwitch).toBeTruthy();

  const component = mount(LSwitch, {
    props: {
      id: 'id',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain(
    'form-control switch switch-md switch-primary'
  );
  expect(component.html()).not.toContain('form-field');
});

test('register css variables', async () => {
  mount(LSwitch);
  expect(
    document.getElementById('lx-theme-provider-global')!.innerText
  ).toContain('--lx-switch');
});

test('inherit attributes', async () => {
  const component = mount(LSwitch, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
