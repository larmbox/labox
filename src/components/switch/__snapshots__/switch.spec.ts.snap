// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"form-control switch switch-md switch-primary\\">
  <div class=\\"switch-inner\\">
    <div><input id=\\"id\\" type=\\"checkbox\\" />
      <div class=\\"switch-box\\">
        <div class=\\"switch-inner-box\\"><span class=\\"icon check switch-icon\\" aria-hidden=\\"true\\"></span><span class=\\"icon x switch-icon\\" aria-hidden=\\"true\\"></span></div>
      </div>
      <!--v-if-->
    </div>
    <div class=\\"switch-context\\">
      <!--v-if-->
      <!--v-if-->
    </div>
  </div>
</div>"
`;
